atlasdb-config/src/main/java/com/palantir/atlasdb/factory/PersistentStoragePathSanitizer.java

```diff
@@ -24,22 +24,30 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
-import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import com.google.common.base.MoreObjects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
 import com.palantir.logsafe.Preconditions;
 import com.palantir.logsafe.SafeArg;
 
-public final class PersistentStorageFactories {
-    private static final Pattern UUID_PATTERN = Pattern.compile(
-            "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
+public final class PersistentStoragePathSanitizer {
+    private static final Logger log = LoggerFactory.getLogger(PersistentStoragePathSanitizer.class);
+    private static final PersistentStoragePathSanitizer INSTANCE = new PersistentStoragePathSanitizer();
+    @VisibleForTesting
+    static final String MAGIC_SUFFIX = "atlasdb-persistent-storage";
 
-    private static final Set<String> SANITIZED_PATHS = new HashSet<>();
+    public static PersistentStoragePathSanitizer create() {
+        return INSTANCE;
+    }
 
-    private PersistentStorageFactories() {}
+    private final Set<String> sanitizedPaths = new HashSet<>();
 
+    @VisibleForTesting
+    PersistentStoragePathSanitizer() {}
```

Comment:
Constructor should be the first method generally, also making it visible for testing is super sketch. We also don't seem to really need a singleton, but if you want to keep it singleton, why don't you make each test use a different storagePath instead?